src/agent/onefuzz-supervisor/src/coordinator.rs

```diff
@@ -202,10 +202,18 @@ impl Coordinator {
         let url = self.registration.dynamic_config.commands_url.clone();
         let request = self.client.get(url).json(&request);
 
-        let pending: PendingNodeCommand = self
-            .send_request(request)
-            .await
-            .context("PollCommands")?
+        let response = self.send_request(request).await.context("PollCommands");
+
+        // Treat communication issues with the service as `no commands
+        // available`.  This adds resilency to the supervisor during longer
+        // outages.  Given poll_commands runs on a 10 second cycle, this should
+        // provide eventual recovery.
+        if let Err(response) = response {
+            error!("error polling the serivce for commands: {:?}", response);
+            return Ok(None);
```

Comment:
I feel uneasy about equivocating "nothing pending" with "can't talk to the service".

Can we define and return a new concrete error type, and have callers inspect that? This would let the caller know that a backoff or even hard failure is warranted after some limit that they can control.

Reply:
addressed